File: Selenium/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://v1.training-support.net/selenium/ajax")
print(driver.title)

driver.find_element(By.XPATH, '//*[@id="ajax-content"]/button').click()

element1 = driver.find_element(By.XPATH, '//*[@id="ajax-content"]/h1')
                     
wait.until(EC.visibility_of(element1))
print(element1.text)

wait.until(EC.text_to_be_present_in_element((By.XPATH, '//*[@id="ajax-content"]/h3'), "I'm late!"))
print(driver.find_element(By.XPATH, '//*[@id="ajax-content"]/h3').text)

driver.quit()

File: Selenium/Activities/Activity9.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://v1.training-support.net/selenium/ajax");
        System.out.println(driver.getTitle());

        WebElement changecontent = driver.findElement(By.cssSelector("button.violet"));
        changecontent.click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.tagName("h1")));
        System.out.println(driver.findElement(By.tagName("h1")).getText());

        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"),"I'm late!"));
        System.out.println(driver.findElement(By.tagName("h3")).getText());

        driver.quit();
    }
}
